cmd/kubeadm/app/util/tokens_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/api"
+)
+
+func TestUsingEmptyTokenFails(t *testing.T) {
+	// Simulates what happens when you omit --token on the CLI
+	config := newConfigWithToken("")
+
+	ok, err := UseGivenTokenIfValid(config)
+	assert.False(t, ok)
+	assert.NoError(t, err)
+}
+
+func TestTokenValidationFailures(t *testing.T) {
+	invalidTokens := []string{
+		"1234567890123456789012",
+		"12345.1234567890123456",
+		".1234567890123456",
+		"123456.1234567890.123456",
+	}
+
+	for _, token := range(invalidTokens) {
+		config := newConfigWithToken(token)
+		ok, err := UseGivenTokenIfValid(config)
+
+		assert.False(t, ok, "expected invalid token to return ok = false: [%s]", token)
+		assert.Error(t, err, "expected invalid token to return an error: [%s]", token)
+		assert.Contains(t, err.Error(), "<6 characters>.<16 characters>", "expected better validation failure message")
```

Comment:
I don't think we need a test to verify you typed what you think you typed.

However, if this method could return different types of errors, those tests would be useful. And generally, inspecting errors strings is a bad idea. They are meant for human consumption. If you need to programatically signal different types of failures through errors, there are better ways. http://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully does a good job of enumerating different approaches to having really useful errors.